### PR TITLE
Add `when`, which allows to modify only a given subtype

### PR DESCRIPTION
**Pull request by stanch**
Example:

```scala
trait Animal
case class Dog(age: Int) extends Animal
case class Cat(ages: List[Int]) extends Animal

case class Zoo(animals: List[Animal])

val zoo = Zoo(List(Dog(4), Cat(List(3, 12, 13))))

val olderZoo = zoo.modifyAll(
  _.animals.each.when[Dog].age,
  _.animals.each.when[Cat].ages.at(0)
).using(_ + 1)
```

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -7,7 +7,8 @@ object BuildSettings {
   val buildSettings = Defaults.coreDefaultSettings ++ Seq (
     organization  := "com.softwaremill.quicklens",
     version       := "1.4.6",
-    crossScalaVersions := Seq("2.11.6", "2.12.0-M3"),
+    scalaVersion  := "2.11.8",
```
Comment (stanch) [inline project/Build.scala]: Without this line SBT falls back to 2.10.x, which prevents the code from compiling :(

**Comment by stanch** [comment]
By the way, this is also known as a prism (http://julien-truffaut.github.io/Monocle//tut/prism.html).

**Comment by adamw** [comment]
Thanks! :) Looks great

**Comment by adamw** [comment]
Released as 1.4.7

**Comment by stanch** [comment]
Awesome, thanks for the fast turnaround (as usual) :)

**Comment by adamw** [comment]
One of these days, I'll got for vacation ;-)

**Pull request merged**